app/api/extension/intellijLogin/route.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  missingParamsPosthogTracking,
+  successPosthogTracking,
+} from "../../../../utils/api/posthogTracking";
+import {
+  missingParamsResponse,
+  successResponse,
+} from "../../../../utils/api/responses";
+import validateParams from "../../../../utils/api/validateParams";
+import intellijLogin from "../../../../utils/db/user/intellijLogin";
+
+export async function POST(request: Request) {
+  const req = await request.json();
+  const { token } = req;
+  const { missingParams } = validateParams(req, ["token"]);
+
+  if (missingParams.length > 0) {
+    missingParamsPosthogTracking({ url: request.url, missingParams });
+    return missingParamsResponse({ missingParams });
+  }
+
+  const userData = await intellijLogin({ watermelon_user: token });
```

Comment:
If a watermelon_user is invalid, the request that returns` {"data": [ ]}` shouldn't return 200.